subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIntegrationTest.groovy

```diff
@@ -1717,6 +1718,51 @@ Found the following transforms:
         output.count("Transforming") == 1
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/6156")
+    def "stops resolving dependencies of task when artifact transforms are encountered"() {
+        given:
+        buildFile << """
+            project(':lib') {
+                task jar(type: Jar) {
+                    destinationDir = buildDir
+                    archiveName = 'lib1.jar'
+                }
+
+                artifacts {
+                    compile jar
+                }
+            }
+
+            project(':app') {
+
+                dependencies {
+                    compile project(':lib')
+                }
+
+                ${configurationAndTransform()}
+
+                task dependent {
+                    dependsOn resolve
+                }
+            }
+
+            gradle.taskGraph.whenReady { taskGraph ->
+                taskGraph.allTasks.each { task ->
+                    task.taskDependencies.getDependencies(task).each { dependency ->
+                        println "> Dependency: \${dependency}"
+                    }
+                }
+            }
+        """
+
+        when:
+        run "dependent"
+
+        then:
+        output.count("> Dependency:") == 1
+        output.contains("> Dependency: task ':app:resolve'")
```

Comment:
Can we check that `resolve` and the artifact transform actually executed?

Reply:
Why is that important for this test? We have plenty of other tests to check this behavior.

Reply:
It is important that the artifact transform and `resolve` are part of the task graph. If they aren't, then the test would not check what we expect it to check.